keps/sig-storage/2848-transfer-volumesnapshots-namespaces/README.md

```diff
@@ -0,0 +1,801 @@
+-->
+This KEP proposes a method for transferring VolumeSnapshot between namespaces. 
+
+Currently, VolumeSnapshot is bound to specific namespace, and VolumeSnapshotContent is Cluster scoped. The binding relationship between these two storage resources is one-to-one.
+This proposal targets allowing users to conveniently move these resources without manual creation
+of each independent object (VolumeSnapshot, VolumeSnapshotContent).
+
+Based on community design sessions, and inspired by [API for PVC Namespace Transfer](https://github.com/kubernetes/enhancements/pull/2110/files), the following is being proposed :-
+
+- `StorageTransferRequest` resources are created in the source namespace.
+- `StorageTransferApproval` resources are created in the target namespace.
+-  When a controller detects matching request/approval resources, a transfer in initiated.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+Allow Kubernetes users to quickly and conveniently transfer VolumeSnapshots across namespaces.
+### Goals
+- Use CRDs for VolumeSnapshot Namespace Transfer request/approval.
+- Edit external SnapshotController to accept/reject/init VolumeSnapshot transfer.
+- The target VolumeSnapshot will point to a newly created VolumeSnapshotContent.
+- Use Validation Webhook to reject any transfer contains secret.
+- Options to be introduced allowing transfer of VolumeSnapshot with secret for specific CSI Drivers (e.g. Ceph).
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Transfer of resources other than VolumeSnapshot will not be discussed.
+- Transfer of PersistentVolumeClaim may be discussed in the future. 
+- Transfer of VolumeSnapshots which contain secret by default.
+
+## Proposal
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+`StorageTransferRequest` resource:
+
+```yaml
+apiVersion: v1alpha1
+kind: StorageTransferRequest
+metadata:
+  name: transfer-foo
+  namespace: target-namespace
+spec:
+  source:
+    name: foo
+    namespace: source-namespace
+    kind: VolumeSnapshot
+  approvalName: approve-foo
+  targetName: bar
```

Comment:
I see that the token/requestToken fields introduced in #2326 are missing.  I thought they added a nice level of addition security